Python/optimizer_bytecodes.c

```diff
@@ -467,6 +467,15 @@ dummy_func(void) {
         res = sym_new_truthiness(ctx, value, false);
     }
 
+    op(_UNARY_NEGATIVE, (value -- res)) {
+        if (sym_matches_type(value, &PyLong_Type) || sym_matches_type(value, &PyFloat_Type)) {
```

Comment:
I think we can add a check for const, and we can setup the res by using `sym_new_const`. Not sure this is worth here cc @brandtbucher @Fidget-Spinner

Reply:
I'd hold off on that, since our plan is to automatically constant-evaluate stuff like this. Let's stick with hand-writing type-based optimizations like this for now.

Reply:
I did not add a const clause for this since it felt more related to #132733.